services/taskcluster-github/index.md

```diff
@@ -0,0 +1,129 @@
+---
+layout:       default
+class:        html
+docson:       true
+marked:       true
+ejs:          true
+superagent:   true
+docref:       true
+---
+
+<h1>TaskCluster GitHub</h1>
+<p>
+Easily trigger TaskCluster jobs based on GitHub pushes and pull requests. Tasks are defind in a YAML coniguration file which lives at the root of a repository. There is no explicit sign-up step, TaskCluster and Mozilla projects will simply begin accepting jobs as soon as a <code>.taskclusterrc</code> exists.
+</p>
+
+### A simple .taskclusterrc file
+<p>
+The following <code>.taskclusterrc</code> will run nodejs tests when a user opens a pull request.
+</p>
+
+```
+---
+version: 1                                     # a version number must be declared
+tasks:                                         # we support multiple tasks
+  - payload:
+      image: "quay.io/mrrrgn/ubuntu-ci:0.0.1"  # some docker image
+      command:                                 # each argument is a new array member
+        - "/bin/bash"
+        - "--login"
+        - "-c"
+        - "checkout-pull-request && npm install . && npm test"
+```
+
+### GitHub Events
+
+You can modify a task definition so that it will only run for specific GitHub events, those events being:
+
+  * pull_request.opened
+  * pull_request.synchronize # a new commit is pushed
+  * pull_request.reopened
+  * pull_request.closed
+  * push                     # a push is made directly to the repo
+
+
+```
+---
+version: 1
+tasks:
+  - payload:
+     image: "ubuntu:latest"
+     command:
+       - "test"
+  - extra:
+      github_events:        # A list of all github events which trigger this task
+        - push
+```
+
+### Task Whitelist Defaults
+<p>
+For security reasons, taskclusterrc jobs are whitelisted by default by user and organization. This means that a task will only be triggered by public members of the organization that own the repository. This behavior may be modified by changing the <code>whitelist:</code> object, defined under <code>extra:</code> in the task definition.
+</p>
+
+```
+---
+version: 1
+tasks:
+  - payload:
+     image: "ubuntu:latest"
+     command:
+       - "test"
+  - extra:
+      whitelist:                # An object containing whitelists
+        users:                  # A list of users allowed to trigger this task
+          - mrrrgn              # github user mrrrgn can now always trigger this task
+          - bob.*               # ALL users with a name that starts: "bob" can trigger this task
```

Comment:
So I recommended yesterday that we remove the whitelist support...
IMO users shouldn't be configured in-tree.. It makes for bad fork-ability and branch-ability...

Suggestions are github teams or some other list maybe... I'm not sure... But I think we should try to come up with a solid solution before we encourage people to adopt it.

One thing is sure though, people with push access have access :)